src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunCrossgen.cs

```diff
@@ -137,18 +143,26 @@ public void It_does_not_support_framework_dependent_publishing(string targetFram
                 "ClassLib");
 
             testProject.AdditionalProperties["PublishReadyToRun"] = "True";
+            testProject.AdditionalProperties["SelfContained"] = "False";
 
             var testProjectInstance = _testAssetsManager.CreateTestProject(testProject)
                 .Restore(Log, testProject.Name);
 
-            // TODO: This test should be changed to expect publishing to succeed when fixing #3109 and #3110.
-            // When fixing the issues, change the function name to reflect what we're testing, and change the test's expected 
-            // behavior (check that the output assemblies exist and are R2R images).
             var publishCommand = new PublishCommand(Log, Path.Combine(testProjectInstance.Path, testProject.Name));
-            publishCommand.Execute("/p:SelfContained=false", "/v:n")
-                .Should()
-                .Fail()
-                .And.HaveStdOutContainingIgnoreCase("NETSDK1095");
+            publishCommand.Execute().Should().Pass();
+
+            DirectoryInfo publishDirectory = publishCommand.GetOutputDirectory(
+                targetFramework,
+                "Debug",
+                EnvironmentInfo.GetCompatibleRid(targetFramework));
```

Comment:
Nit: Don't call `EnvironmentInfo.GetCompatibleRid` twice here.  Use `testProject.RuntimeIdentifier`, or store the result of the first call in a variable.

Reply:
Fixed.